### PR TITLE
Map source url fragment to path.

### PR DESCRIPTION
**Pull request by n1k0**
This patch updates routing url parsing to map the source fragment as the internal URL path, so query strings are handled properly.

It also upgrades a bunch of dependency and fixes the *elm-analyse* errors in skeleton.  

These are mostly breaking changes, so we bump a new 5.0.0 version.

### CODE REVIEW

**Review by fbentz — APPROVED** [review]
:+1: Can you release a new package?

File: skeleton/package.json [hunk 1]
```diff
@@ -18,13 +18,13 @@
     "params": "pure_funcs=\"F2,F3,F4,F5,F6,F7,F8,F9,A2,A3,A4,A5,A6,A7,A8,A9\",pure_getters,keep_fargs=false,unsafe_comps,unsafe"
   },
   "devDependencies": {
-    "elm": "^0.19.0-bugfix6",
+    "elm": "^0.19.0-no-deps",
```
Comment (Natim) [inline skeleton/package.json]: Never heard of this, is it a new thing?

Comment (n1k0) [reply skeleton/package.json]: That's the latest stable version available from npm.

**Comment by n1k0** [comment]
Published as v5.0.0.

**Pull request merged**